packages/components-react/icon-react/src/icon.tsx

```diff
@@ -0,0 +1,45 @@
+import type { HTMLAttributes } from 'react';
+import { clsx } from 'clsx';
+import { forwardRef } from 'react';
+
+type Dir = HTMLAttributes<HTMLSpanElement>['dir'];
+
+export type IconProps =
+  | (Omit<HTMLAttributes<HTMLSpanElement>, 'role' | 'aria-hidden' | 'aria-label' | 'aria-labelledby' | 'dir'> & {
+      role?: never;
+      'aria-hidden'?: true;
+      'aria-label'?: never;
+      'aria-labelledby'?: never;
+      dir?: Dir;
+    })
+  | (Omit<HTMLAttributes<HTMLSpanElement>, 'role' | 'aria-hidden'> & {
+      role: 'img';
+      'aria-label': string;
+      'aria-labelledby'?: never;
+      'aria-hidden'?: never;
+      dir?: Dir;
+    })
+  | (Omit<HTMLAttributes<HTMLSpanElement>, 'role' | 'aria-hidden'> & {
+      role: 'img';
+      'aria-labelledby': string;
+      'aria-label'?: never;
+      'aria-hidden'?: never;
+      dir?: Dir;
+    });
+
+function hasRoleImg(props: IconProps): props is Extract<IconProps, { role: 'img' }> {
+  return props.role === 'img';
+}
+
+export const Icon = forwardRef<HTMLSpanElement, IconProps>(function Icon(props, forwardedRef) {
+  const { children, className, dir, ...restProps } = props;
```

Comment:
Waarom extract je de `dir` property? Er wordt niets mee gedaan toch?

Reply:
Eigenlijk alleen om het expliciete deel van de API duidelijk te maken maar dat doe ik ook niet met `role` dus ik haal hem wel weg.

Reply:
Ik heb ze wel in het props type laten staan om de API helder te hebben.